includes/Checker/Checks/Enqueued_Scripts_Size_Check.php

```diff
@@ -0,0 +1,196 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Checks\Enqueued_Scripts_Size_Check
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker\Checks;
+
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Checker\With_Shared_Preparations;
+use WordPress\Plugin_Check\Traits\URL_Aware;
+use Exception;
+
+/**
+ * Check for running WordPress internationalization sniffs.
+ *
+ * @since n.e.x.t
+ */
+class Enqueued_Scripts_Size_Check extends Abstract_Runtime_Check implements With_Shared_Preparations {
+
+	use URL_Aware;
+
+	/**
+	 * Threshold for script size to surface a warning for.
+	 *
+	 * @since 1.0.0
+	 * @var int
+	 */
+	public $threshold_size = 300000;
```

Comment:
Thanks @felixarntz I've updated this and other properties/methods to private/protected where reasonable.

I have used your idea with a class constructor for the threshold as this would help with testing.